### PR TITLE
linter: summarize status at end

### LINKED ISSUE #571: linter: summarize results at the end
the linter's output can be difficult to review because it is so long. key information, like if there was a failure and which rules failed, can be hidden halfway through the output and not visible at the end. we should summarize the results of linting at the conclusion of the execution.

### PR DESCRIPTION
**Pull request by williballenthin**
closes #571

### CODE REVIEW

**Comment by williballenthin** [comment]
image

**Review by mr-tz — COMMENTED** [review]
nice, just one semi-related question while we're at it

also needs `black` reformatting

**Review by Ana06 — APPROVED** [review]
This is great!!! 🚀  It needs to run `black` and to to rebase (to solve the merge conflicts in the CHANGELOG).

File: scripts/lint.py [hunk 1]
```diff
@@ -528,15 +528,16 @@ def lint_rule(ctx, rule):
 
         print("")
 
-    lints_failed = any(map(lambda v: v.level == Lint.FAIL, violations))
+    lints_failed = len(tuple(filter(lambda v: v.level == Lint.FAIL and not is_nursery_rule(rule), violations)))
```
[the remark this answers is not in the document]
Comment (williballenthin) [reply scripts/lint.py:531]: ok, but won't fail on no example. any other exceptions?

[the remark this answers is not in the document]
Comment (mr-tz) [reply scripts/lint.py]: that should be it

**Review by mr-tz — APPROVED** [review]
thanks!

**Comment by mr-tz** [comment]
Looks like the CHANGELOG update got lost. Please add and then merge.

File: scripts/lint.py [hunk 2]
```diff
@@ -677,16 +704,33 @@ def main(argv=None):
         "is_thorough": args.thorough,
     }
 
-    did_violate = lint(ctx, rules)
+    results_by_name = lint(ctx, rules)
+    failed_rules = []
+    warned_rules = []
+    for name, (fail_count, warn_count) in results_by_name.items():
```
Comment (mike-hunhoff) [inline scripts/lint.py:710]: TIL

**Review by mike-hunhoff — APPROVED** [review]
awesome work, @williballenthin! LGTM

**Pull request merged**